File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/converter/TripWithDetailsToTripConverter.java
package com.worldventures.dreamtrips.modules.trips.model.converter;

import com.worldventures.dreamtrips.api.trip.model.TripWithDetails;
import com.worldventures.dreamtrips.modules.trips.model.ContentItem;
import com.worldventures.dreamtrips.modules.trips.model.TripModel;

import io.techery.mappery.MapperyContext;

public class TripWithDetailsToTripConverter extends ApiTripToTripConverter<TripWithDetails> {

   @Override
   public Class<TripWithDetails> sourceClass() {
      return TripWithDetails.class;
   }

   @Override
   public TripModel convert(MapperyContext mapperyContext, TripWithDetails tripWithDetails) {
      TripModel tripModel = convertTrip(mapperyContext, tripWithDetails);
      tripModel.setContent(mapperyContext.convert(tripWithDetails.contentItems(), ContentItem.class));
      return tripModel;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/converter/TripWithDetailsToTripConverter.kt
package com.worldventures.dreamtrips.modules.trips.model.converter

import com.worldventures.dreamtrips.api.trip.model.TripWithDetails
import com.worldventures.dreamtrips.modules.trips.model.ContentItem

import io.techery.mappery.MapperyContext

class TripWithDetailsToTripConverter : ApiTripToTripConverter<TripWithDetails>() {

   override fun sourceClass() = TripWithDetails::class.java

   override fun convert(context: MapperyContext, source: TripWithDetails) = convertTrip(context, source)
         .apply { content = context.convert(source.contentItems(), ContentItem::class.java) }
}
